lib/shellable.ts

```diff
@@ -369,13 +418,51 @@ export class LinuxPlatform extends ShellPlatform {
     lines.push(`unzip /tmp/$(basename \$${S3_KEY_ENV}) -d /tmp/scriptdir`);
 
     if (assumeRole) {
-      const externalId = assumeRole.externalId ? `--external-id "${assumeRole.externalId}"` : '';
-      const StsEndpoints = useRegionalStsEndpoints ? "regional" : "legacy";
-      lines.push('creds=$(mktemp -d)/creds.json');
-      lines.push(`AWS_STS_REGIONAL_ENDPOINTS=${StsEndpoints} aws sts assume-role --role-arn "${assumeRole.roleArn}" --role-session-name "${assumeRole.sessionName}" ${externalId} > $creds`);
-      lines.push('export AWS_ACCESS_KEY_ID="$(cat ${creds} | grep "AccessKeyId" | cut -d\'"\' -f 4)"');
-      lines.push('export AWS_SECRET_ACCESS_KEY="$(cat ${creds} | grep "SecretAccessKey" | cut -d\'"\' -f 4)"');
-      lines.push('export AWS_SESSION_TOKEN="$(cat ${creds} | grep "SessionToken" | cut -d\'"\' -f 4)"');
+
+      if (assumeRole.profileName) {
+
+        if (!assumeRole.credentialSource && !assumeRole.sourceProfile) {
+          throw new Error(`Either 'credentialSource' or 'sourceProfile' is required when using a profile`);
+        }
+
+        if (assumeRole.credentialSource && assumeRole.sourceProfile) {
+          throw new Error(`Only one of 'credentialSource' or 'sourceProfile' is allowed when using a profile`);
+        }
+
+        const awsHome = assumeRole.awsHomeDir ?? '~/.aws';
+
+        lines.push(`mkdir -p ${awsHome}`);
+        lines.push(`touch ${awsHome}/credentials`);
+        lines.push(`config=${awsHome}/config`);
+        lines.push(`echo [profile ${assumeRole.profileName}]>> $\{config\}`);
+
+        if (assumeRole.credentialSource) {
+          lines.push(`echo credential_source = ${assumeRole.credentialSource} >> $\{config\}`);
+        } else {
+          lines.push(`echo source_profile = ${assumeRole.sourceProfile} >> $\{config\}`);
+        }
+
+        lines.push(`echo role_session_name = ${assumeRole.sessionName} >> $\{config\}`);
+        lines.push(`echo role_arn = ${assumeRole.roleArn} >> $config`);
+
+        if (assumeRole.externalId) {
+          lines.push(`echo external_id = ${assumeRole.externalId} >> $config`);
+        }
+
+        // let the application code know which role is being used.
+        lines.push(`export AWS_PROFILE=${assumeRole.profileName}`);
```

Comment:
Ah yeah I guess that works too.

Reply:
I'm in doubt whether we should set `AWS_SDK_LOAD_CONFIG=1` as well here.

In *principle* that's the JavaScript SDK user's responsibility (AWS CLI doesn't need it for example)... but it feels like a good safeguard against misconfiguration and annoyance downstream.